IoTEdgeModbus/Modules/AzureIoTEdgeModbus/Instrumentation/MicrosoftExtensionsLog.cs

```diff
@@ -33,31 +33,31 @@ public void IoTHubConnectionOpened()
             this.LogInformation(MethodBase.GetCurrentMethod());
         }
         
-        [Event(2002, Description = "New desired properties received: {0}.")]
+        [Event(2002, Description = "New desired properties received: {desiredProperties}.")]
         public void DesiredPropertiesReceivedFromTwin(TwinCollection twinCollection)
         {
             this.LogInformation(MethodBase.GetCurrentMethod(), twinCollection);
         }
 
-        [Event(2003, Description = "New desired properties received: {0}.")]
+        [Event(2003, Description = "New desired properties received: {desiredProperties}.")]
         public void DesiredPropertiesReceivedFromFile(string desiredProperties)
         {
             this.LogInformation(MethodBase.GetCurrentMethod(), desiredProperties);
         }
 
-        [Event(2004, Description = "Attempting to retrieve desired configuration from provider: {0}.")]
+        [Event(2004, Description = "Attempting to retrieve desired configuration from provider: {providerType}.")]
         public void RetrieveDesiredConfigurationFrom(string configurationProviderTypeName)
         {
             this.LogInformation(MethodBase.GetCurrentMethod(), configurationProviderTypeName);
```

Comment:
Do we want to use reflection here? Consider using nameof(RetrieveDesiredConfigurationFrom) instead. All methods in the class seems to use reflection so in case we change we should change all of them.

Reply:
That will not work IMHO, as methodbase returns MemberInfo which is used in base class to extract custom attribute properties. I appreciate the use reflection is subject to scrutiny but in this case we are not using it for high performance telemetry, it's more event based. May be we can have another stab at it if we start to use it for high performance logging and do some load testing. As a stop gap I may add caching of MemberInfo if we run into any performance issues, I suspect not.

Reply:
My thought was just regarding the presence of "LogTrace(MemberInfo mi, params object[] args)" and that it might be used in loops where performance could be crucial. As you stated it is not right now, so I mark as resolved. In case we want to change we would have to change the attribute handling as well. Lets get it to master. :)